x-pack/plugins/security_solution/server/lib/detection_engine/rule_types/query/query.ts

```diff
@@ -58,7 +58,6 @@ export const queryExecutor = async ({
       services,
       index: runOpts.inputIndex,
       exceptionFilter: runOpts.exceptionFilter,
-      fields: runOpts.inputIndexFields,
```

Comment:
It might be easier to create a new copy of the `getFilter` function that integrates the data view fetching logic instead of modifying the existing one. We can use the new one only here for Query rules so it doesn't affect the rest of the rules which are much less common than Query and EQL rules.